### PR TITLE
Reinforce CI

### PR DESCRIPTION
**Pull request by ShigekiKarita**
- [x] fix shellcheck 0.6.0 (it's stronger)
- [x] travis: add chainer 6.0.0 (see #753 #760 #761 ), pytorch 1.1.0 and nightly. 
- [x] travis: remove pytorch 0.4.1
- [x] travis: update xenial
- [x] circle ci: docker ubuntu 16/18 and centos7(+devtoolset-7) images https://circleci.com/gh/espnet/espnet/tree/travis-nightly

this is an essential step to exterminate build errors and to release prebuilt binary.

### CODE REVIEW

**Comment by ShigekiKarita** [comment]
shellcheck becomes much stronger in xenial! please help https://travis-ci.org/espnet/espnet/jobs/537357918

**Comment by ShigekiKarita** [comment]
@kamo-naoyuki Can you fix this warning? I have no idea even it should be fixed or not.
```
$ ./ci/test_shell.sh
=> utils/eval_source_separation.sh
In ./eval_source_separation.sh line 33:
reffiles=( $(echo $1 | tr , " ") )
           ^-------------------^ SC2207: Prefer mapfile or read -a to split command output (or quote to avoid splitting).


In ./eval_source_separation.sh line 34:
enhfiles=( $(echo $2 | tr , " ") )
           ^-------------------^ SC2207: Prefer mapfile or read -a to split command output (or quote to avoid splitting).
```

**Comment by ShigekiKarita** [comment]
shellcheck report is fixed everything but SC2207

**Comment by ShigekiKarita** [comment]
I will split common blocks in .travis.yml into `ci/` for the other CI. I plan to support Circle CI because it supports [docker](https://circleci.com/docs/2.0/executor-intro/#section=configuration) and [ssh debug](https://circleci.com/docs/2.0/ssh-access-jobs/). At least ubuntu16/18, and centos7 images will be test.

Any requests or recommendations are welcome.

**Comment by kamo-naoyuki** [comment]
```bash
IFS=, read -a reffiles <<<$1
IFS=, read -a enhfiles <<<$2 
```
Maybe it's better to follow shellcheck.

**Comment by ShigekiKarita** [comment]
Thanks!!

**Comment by ShigekiKarita** [comment]
@kamo-naoyuki I've added `-r` because shellcheck said that
> SC2162: read without -r will mangle backslashes.
```bash
IFS=, read -r -a reffiles <<<$1
IFS=, read -r -a enhfiles <<<$2 
```

**Comment by ShigekiKarita** [comment]
I reduced data dim in `test_sortagrad_tranable` because it took an unreasonably long time (5-10 min?).

**Comment by ShigekiKarita** [comment]
DONE!! Can anyone review this? It is not a big change. Now we can support people using CentOS7 and Ubuntu16/18 strictly.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            yum install -y git centos-release-scl cmake libsndfile make bzip2 wget which unzip
+            yum-config-manager --enable rhel-server-rhscl-7-rpms
+            yum install -y devtoolset-7-gcc-c++
+      - checkout
+      - run:
+          name: run tests
+          command: |
+            # NOTE(karita) this line exited 1
+            # source scl_source enable devtoolset-7
+            PATH="/opt/rh/devtoolset-7/root/usr/bin:${PATH:-}"
```
Comment (kan-bayashi) [inline .circleci/config.yml]: This line is used instead of  `source scl_source enable devtoolset-7` ? 
(I want to just confirm)

**Comment by ShigekiKarita** [comment]
> If the error is happened with nightly version, the check goes through or not?

Currently no. I will `allow_failures` for nightly. Just a moment.

File: .circleci/config.yml (hunk 1 above)
Comment (ShigekiKarita) [reply .circleci/config.yml]: Yes. This is very strange. When I ssh into circle ci, it accepts `source scl_source enable devtoolset-7` but `config.yml` does not.

Comment (kan-bayashi) [reply .circleci/config.yml]: :(

**Comment by ShigekiKarita** [comment]
Now we can see the job with pytorch nightly or chainer alpha are allowed to fail https://travis-ci.org/espnet/espnet/builds/537604830

File: .circleci/config.yml (hunk 1 above)
Comment (ShigekiKarita) [reply .circleci/config.yml]: Anyone can retry that by "Rerun workflow > Rerun job with ssh" https://circleci.com/gh/espnet/espnet/59

**Comment by ShigekiKarita** [comment]
Done

**Comment by ShigekiKarita** [comment]
At this time, I cannot find Ubuntu16/18 and CentOS7 failed to build warp-ctc or something. Does anyone (@sw005320, @jnishi) request more OS to test?

**Comment by sw005320** [comment]
JHU CLSP uses Debian 9.4 but I think Ubuntu16/18 supports should be fine.

**Comment by sw005320** [comment]
Can we also add different GCC (we use 6.3.0)? It is no problem if this makes it complicated.

**Comment by ShigekiKarita** [comment]
OK

**Comment by ShigekiKarita** [comment]
Done

**Pull request merged**